SourcingTests/CollectionViewDataSourceTest.swift

```diff
@@ -182,5 +182,21 @@ class CollectionViewDataSourceTest: XCTestCase {
         XCTAssertEqual(secondCollectionViewMock.reloadedCount, 1)
     }
 
-
+    func testMoveIndexPaths() {
+        //Given
+        let cellConfig = CellConfiguration<MockCollectionCell<Int>>(cellIdentifier: cellIdentifier)
+        let realCollectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewLayout())
+        
+        //When
+        let dataSource = CollectionViewDataSource(collectionView: collectionViewMock, dataProvider: dataProvider, cellDequeable: cellConfig)
+        let fromIndexPath = IndexPath(row: 0, section: 1)
+        let toIndexPath = IndexPath(row: 1, section: 0)
+        dataProvider.moveItemAt(sourceIndexPath: fromIndexPath, to: toIndexPath)
+        let rowCountFirstSection = dataSource.collectionView(realCollectionView, numberOfItemsInSection: 0)
```

Comment:
Think about what we are testing here. We want to test the dataSource. But wer are testing dataProvider in here as well. This can be dangerous, because the dataProvider is doing the moving logic. If dataProvider fails this would cause this test to fail. This should not happen